### ISSUE TITLE
Graphics card crashes when trying to start Waydroid

### ISSUE THREAD

**Issue by HanoJing** [waydroid/waydroid#860]
**Describe the bug**
The Graphics card (an AMD Radeon RX580 XFX in my case) is crashing when trying to start waydroid.

**General information (please complete the following information):**
 - Waydroid tools Version: 1.3.4
 - Waydroid Images Version: `lineage-18.1-20230422-VANILLA-waydroid_arm64-system`
                                                    `lineage-18.1-20230422-MAINLINE-waydroid_arm64-vendor`

**Desktop (please complete the following information):**
 - OS: Void Linux ARM64
 - GPU: AMD Radeon RX580 XFX
 - Kernel version: 6.1.3
 - Host mesa version: 22.3.5
 - Desktop: Hyprland, Wayland

**Additional context**
Maybe another AMDGPU driver issue
On my RockPro64, waydroid is starting up just fine.

**Logs (please upload as file)**
- Prop file: [waydroid-prop.log](https://github.com/waydroid/waydroid/files/11300343/waydroid-prop.log)
- Log file: [waydroid.log](https://github.com/waydroid/waydroid/files/11300344/waydroid.log)
- Waydroid logcat: [waydroid-logcat.log](https://github.com/waydroid/waydroid/files/11300345/waydroid-logcat.log)
- dmesg: [dmesg.log](https://github.com/waydroid/waydroid/files/11300346/dmesg.log)

**Comment by aleasto** [comment]
Did `lineage-18.1-20230415-MAINLINE-waydroid_arm64-vendor` work? The difference would be `mesa-23.0.1` vs `mesa-23.0.3`

I don't have issues with AMD on x86_64, on either version

**Comment by HanoJing** [comment]
> Did `lineage-18.1-20230415-MAINLINE-waydroid_arm64-vendor` work? The difference would be `mesa-23.0.1` vs `mesa-23.0.3`
> 
> I don't have issues with AMD on x86_64, on either version

Yes, 0415 did work.

**Comment by aleasto** [comment]
Ok. You should try mesa-23.0.3 on your host and report a bug upstream

**Comment by HanoJing** [comment]
Alright, I now confirmed that having Mesa 23.0.3 as the host version also doesn't fix the issue, same goes for mesa-upstream as the host version. 

Here is the Gitlab issue: https://gitlab.freedesktop.org/mesa/mesa/-/issues/8890

**Comment by aleasto** [comment]
well, i was thinking that mesa 23.0.3 would break your host like it broke waydroid.

**Comment by HanoJing** [comment]
> well, i was thinking that mesa 23.0.3 would break your host like it broke waydroid.

Oh, It might be that only Android is affected here.

**Comment by HanoJing** [comment]
@aleasto Hey back, since you told me that you're not having this issue, are you using an Nvidia GPU? I looked some more at those crashes and this bug seems to be caused on AMD GPUs. Especially the message: [drm:amdgpu_cs_ioctl [amdgpu]] *ERROR* Failed to initialize parser -125!
This bug has existed for a long time already and many people are having it: https://bugzilla.kernel.org/show_bug.cgi?id=205089


When Waydroid crashes there is a VM fault like here though:
https://gitlab.freedesktop.org/mesa/mesa/-/issues/5462

I guess I'll get this one fixed for my system next up, somehow.

**Comment by niksingh710** [comment]
![image](https://github.com/waydroid/waydroid/assets/60490474/08a90462-71bb-4e3b-906f-2bc455473630)
I see this if i force waydroid to use my amdgpu.

GPU: AMD ATI Radeon R7 M260/M265 / M340/M360 / M440/M445 / 530/535 / 620/625 Mobile

**Comment by HanoJing** [comment]
I see not the exact same glitches but similar ones. 
And in my case the graphics driver crashes arbitrarily and causes a similar distortion to show up after a few seconds 
of a frozen screen.

**Comment by HanoJing** [comment]
> > well, i was thinking that mesa 23.0.3 would break your host like it broke waydroid.
> 
> Oh, It might be that only Android is affected here. Or Vulkan maybe. I'll do some tests with upstream host mesa.
> 
> Update: I have tested the following applications:
> 
>     * Minecraft
> 
>     * minetest
> 
>     * teeworlds
> 
>     * supertuxkart
> 
>     * supertux2
> 
>     * redeclipse: the only application that triggered an AMDGPU driver crash.
> 
>     * xonotic
> 
>     * neverball
> 
>     * veloren
> 
>     * rvgl
> 
>     * speed-dreams
> 
>     * OBS-Studio

Regarding redeclipse, here's what the dmesg looks like: [dmesg-normal.log](https://github.com/waydroid/waydroid/files/11778319/dmesg-normal.log)

**Comment by HanoJing** [comment]
This bug got fixed with the latest 20230723 images!

**Comment by niksingh710** [comment]
> This bug got fixed with the latest 20230723 images!

waydroid init will get this image or have to manually install them?

**Comment by HanoJing** [comment]
> > This bug got fixed with the latest 20230723 images!
> 
> waydroid init will get this image or have to manually install them?

Waydroid init will get this image.

**Comment by xiaohuirong** [comment]
I have an AMD rx580 graphics card. When using the `18.1_20240224` image, I encountered a similar issue where the following three programs kept crashing: 
```
/vendor/bin/hw/android.hardware.graphics.allocator@2.0-service
/vendor/bin/hw/android.hardware.graphics.composer@2.1-service
/system/bin/surfaceflinger
```
This prevented Waydroid from starting. However, after downgrading to the `20230723` version image, everything worked fine.

/var/lib/waydroid/lxc/waydroid/config_nodes:
```
lxc.mount.entry = tmpfs dev tmpfs nosuid 0 0
lxc.mount.entry = /dev/zero dev/zero none bind,create=file,optional 0 0
lxc.mount.entry = /dev/null dev/null none bind,create=file,optional 0 0
lxc.mount.entry = /dev/full dev/full none bind,create=file,optional 0 0
lxc.mount.entry = /dev/fuse dev/fuse none bind,create=file,optional 0 0
lxc.mount.entry = /dev/tty dev/tty none bind,create=file,optional 0 0
lxc.mount.entry = /dev/char dev/char none bind,create=dir,optional 0 0
lxc.mount.entry = /dev/fb0 dev/fb0 none bind,create=file,optional 0 0
lxc.mount.entry = /dev/anbox-binder dev/binder none bind,create=file,optional 0 0
lxc.mount.entry = /dev/anbox-vndbinder dev/vndbinder none bind,create=file,optional 0 0
lxc.mount.entry = /dev/anbox-hwbinder dev/hwbinder none bind,create=file,optional 0 0
lxc.mount.entry = none dev/pts devpts defaults,mode=644,ptmxmode=666,create=dir 0 0
lxc.mount.entry = /dev/uhid dev/uhid none bind,create=file,optional 0 0
lxc.mount.entry = /dev/net/tun dev/tun none bind,create=file,optional 0 0
lxc.mount.entry = /var/lib/waydroid/host-permissions vendor/etc/host-permissions none bind,optional 0 0
lxc.mount.entry = /sys/kernel/debug sys/kernel/debug none rbind,create=dir,optional 0 0
lxc.mount.entry = tmpfs mnt_extra tmpfs nodev 0 0
lxc.mount.entry = tmpfs tmp tmpfs nodev 0 0
lxc.mount.entry = tmpfs var tmpfs nodev 0 0
lxc.mount.entry = tmpfs run tmpfs nodev 0 0
lxc.mount.entry = /dev/dri/card2 dev/dri/card0 none bind,create=file,optional 0 0
lxc.mount.entry = /dev/dri/renderD129 dev/dri/renderD128 none bind,create=file,optional 0 0

```
/var/lib/waydroid/waydroid_base.prop:
```
sys.use_memfd=true
debug.stagefright.ccodec=0
ro.hardware.gralloc=gbm
ro.hardware.egl=mesa
ro.hardware.vulkan=radeon
ro.hardware.camera=v4l2
ro.opengles.version=196609
waydroid.updater.disabled=true
waydroid.tools_version=1.4.1
ro.vndk.lite=true

```
